base/interactiveutil.jl

```diff
@@ -23,31 +23,35 @@ function editor()
     return command
 end
 
-function edit(file::AbstractString, line::Integer)
+function edit(path::AbstractString, line::Integer=1)
     command = editor()
     name = basename(first(command))
-    issrc = length(file)>2 && file[end-2:end] == ".jl"
+    issrc = length(path)>2 && path[end-2:end] == ".jl"
     if issrc
-        f = find_source_file(file)
-        f !== nothing && (file = f)
+        f = find_source_file(path)
+        f !== nothing && (path = f)
     end
     const no_line_msg = "Unknown editor: no line number information passed.\nThe method is defined at line $line."
     if startswith(name, "emacs") || name == "gedit"
-        spawn(`$command +$line $file`)
+        args = isfile(path) ? `+$line $path` : `$path`
```

Comment:
I guess this works, but it seems cleaner to me to use an array here for `args`, i.e. `["+$line" path] : [path]`.